src/application.rs

```diff
@@ -1,129 +1,45 @@
-use std::iter;
 use std::sync::Arc;
-use std::time::SystemTime;
+use std::time::{Duration, SystemTime};
 
-use leptos::{ReadSignal, WriteSignal};
+use leptos::WriteSignal;
 use winit::application::ApplicationHandler;
-use winit::event::{DeviceEvent, DeviceId, KeyEvent, WindowEvent};
-use winit::event_loop::{ActiveEventLoop, EventLoopClosed, EventLoopProxy};
-use winit::keyboard::Key;
+use winit::event::{KeyEvent, WindowEvent};
+use winit::event_loop::{ActiveEventLoop, ControlFlow, EventLoopProxy};
+use winit::keyboard::{KeyCode, PhysicalKey};
 use winit::window::{WindowAttributes, WindowId};
 
-use control_overlay::ControlState;
-
 use crate::build::tenscript::{FabricPlan, TenscriptError};
 use crate::build::tenscript::brick_library::BrickLibrary;
 use crate::build::tenscript::fabric_library::FabricLibrary;
 use crate::camera::Pick;
-use crate::control_overlay;
-use crate::control_overlay::action::Action;
+use crate::control_overlay::menu::{EventMap, MenuContent};
 use crate::crucible::{Crucible, CrucibleAction};
+use crate::messages::{ControlState, LabEvent, SceneAction};
 use crate::scene::Scene;
-use crate::wgpu_context::WgpuContext;
+use crate::wgpu::Wgpu;
 
-pub struct Application<'window> {
+pub struct Application {
```

Comment:
oooh new version of wgpu means you don't have to pass in lifetimes to scene anymore?